eZ/Publish/Core/Repository/URLWildcardService.php

```diff
@@ -82,14 +78,11 @@ public function create($sourceUrl, $destinationUrl, $forward = false)
         $sourceUrl = $this->cleanUrl($sourceUrl);
         $destinationUrl = $this->cleanUrl($destinationUrl);
 
-        $spiUrlWildcards = $this->urlWildcardHandler->loadAll();
-        foreach ($spiUrlWildcards as $wildcard) {
-            if ($wildcard->sourceUrl === $sourceUrl) {
-                throw new InvalidArgumentException(
-                    '$sourceUrl',
-                    'Pattern already exists'
-                );
-            }
+        if ($this->urlWildcardHandler->exists($this->cleanPath($sourceUrl))) {
```

Comment:
Exists is prone to race conditions if people for some reason end up doing exists + load call.
Would it make sense rather do a translate() here and enhance the exception with the info so it becomes `Pattern already exists and currently points to <target>`

In order to be able to drop `exists()` 😅

Reply:
EDIT: I see exists has different semantic now, it only looks for exact match and not wildcards.

Maybe rename to `exactSourceUrlExists()`, aka something more specific? _(or something better, check in the office what others think)_

either way also phpdoc should better reflect this.

Reply:
I changed the method name to the proposed one. I hope the intentions are more clear now.